### ISSUE TITLE
Combine multiple filters

### ISSUE THREAD

**Issue by kentsay** [yahoo/elide#337]
Hi guys, 

Got some questions about using `filter` in Elide. 

The first one is regarding combining multiple filters:
According to jsonapi [recommendations](http://jsonapi.org/recommendations/), multiple filters can be applied to a single request. For example: 
```
GET /comments?filter[post]=1,2&filter[author]=12 HTTP/1.1
```
The document I found on Elide's website is also the same:
```
/author/1/book?filter[book.genre]=Science%20Fiction,filter[book.title][prefix]=The
```

But it doesn't work as it states to combine multiple filters with `comma`, instead, it seems you need to combine them with `&`:
```
/api/v1/people?filter[people.firstName][infix]=F&filter[people.lastName][infix]=W
```

My second question is: Is there something like 'OR' operator in Elide when combining multiple filter?

Can anyone help to clarify this? Thanks.

**Comment by DeathByTape** [comment]
hi @kentsay, yes it looks like that is a mistake in our documentation, thanks for pointing this out!

As far as the `OR` operator goes, I know we had some issues in our original filtering scheme but I think @aklish had added a new scheme which may not have documentation yet. It should follow the [FIQL](https://github.com/jirutka/rsql-parser) spec. 

@aklish or @wcekan do you remember if we had gotten this merged?

**Comment by aklish** [comment]
Hi.  A couple of points:

First, it looks like you found a bug in the documentation.  I'll work on resolving that.  

Elide supports multiple filter dialects, but they have to be enabled.  It includes two dialects in the core code - _basic_ - which is similar to JSON-API recommendations and _RSQL_ - which allows full filter boolean expressions.

Here is a pointer to some Elide tests that illustrate different examples of these filters in action.

Basic:
https://github.com/yahoo/elide/blob/1586d264ccd41a285f11d784930dbe7c09db457b/elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/FilterIT.groovy#L1385

Here is the exact same expression with RSQL (Notice ';' is used for conjunction instead of using '&'):
https://github.com/yahoo/elide/blob/1586d264ccd41a285f11d784930dbe7c09db457b/elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/FilterIT.groovy#L1391

Elide supports 'OR' using _RSQL_ (replace ',' for ';' in the 2nd example above).  'OR' isn't supported in _basic_ except in a limited sense by having the 'in' clause.

Elide by default doesn't have RSQL enabled in any of the examples.  It must be explicitly turned on in code with a one line change.  In Elide 3.0, it will likely be available by default.

**Comment by aklish** [comment]
Here is an example of the IT tests enabling both _basic_ and _RSQL_ filter dialects:

https://github.com/yahoo/elide/blob/1586d264ccd41a285f11d784930dbe7c09db457b/elide-integration-tests/src/test/java/com/yahoo/elide/initialization/StandardTestBinder.java#L39

all the way through line 46.

**Comment by kentsay** [comment]
Hi @aklish, thanks for the examples. 
Please close this issue after you update the document.

**Comment by clayreimann** [comment]
Closed by yahoo/elide-doc#38
